lsp-semantic-tokens.el

```diff
@@ -541,6 +577,144 @@ LOUDLY will be forwarded to OLD-FONTIFY-REGION as-is."
+end of the comment pair that contains END. If either of such
+comment pairs does not exist, return nil for that part. This
+function also removes dangling comment starters/ends."
+  (let (prev-beg
+        prev-end
+        next-beg
+        next-end
+        tmp)
+    (lsp-save-restriction-and-excursion
+      (goto-char beg)
+      (setq prev-beg (text-property-search-backward 'lsp-semantic-token--comment-beg))
+      (when prev-beg
+        (setq prev-beg (prop-match-beginning prev-beg))
+        (goto-char prev-beg)
+        (setq prev-end (text-property-search-forward 'lsp-semantic-token--comment-end))
+        (when prev-end
+          (setq prev-end (prop-match-end prev-end))
+          ;; Check whether this is the actual matching end of prev-start
+          (setq tmp (text-property-search-backward 'lsp-semantic-token--comment-beg))
+          (when (or (not tmp) (not (equal prev-beg (prop-match-beginning tmp))))
+            (setq prev-end nil)))
+
+        (if prev-end
+            (when (<= prev-end beg)
+              ;; Does not overlap with (beg end)
+              (setq prev-beg nil))
+          (lsp-semantic-tokens--remove-comment-syntax-strict prev-beg prev-beg)
+          (setq prev-beg nil)))
+
+      (goto-char end)
+      (setq next-end (text-property-search-forward 'lsp-semantic-token--comment-end))
+      (when next-end
+        (setq next-end (prop-match-end next-end))
+        (goto-char next-end)
+        (setq next-beg (text-property-search-backward 'lsp-semantic-token--comment-beg))
+        (when next-beg
+          (setq next-beg (prop-match-beginning next-beg))
+          ;; Check whether this is the actual matching beginning of next-end
+          (setq tmp (text-property-search-forward 'lsp-semantic-token--comment-end))
+          (when (or (not tmp) (not (equal next-end (prop-match-end tmp))))
+            (setq next-beg nil)))
+
+        (if next-beg
+            (when (>= next-beg end)
+              ;; Does not overlap with (beg end)
+              (setq next-end nil))
+          (lsp-semantic-tokens--remove-comment-syntax-strict next-end next-end)
+          (setq next-end nil))))
+    (cons prev-beg next-end)))
+
+(defun lsp-semantic-tokens--remove-comment-syntax-strict (beg end)
+  "Remove all commnet syntax strictly in (BEG END), even if they
+overlap out of the range."
+  (lsp-save-restriction-and-excursion
+    (with-silent-modifications
+      ;; Remove comment starters
+      (goto-char beg)
+      (cl-do ((loc (text-property-search-forward
+                    'lsp-semantic-token--comment-beg)
+                   (text-property-search-forward
+                    'lsp-semantic-token--comment-beg)))
+          ((or (not loc) (>= (point) end)))
+        (let ((beg-match (prop-match-beginning loc))
+              (end-match (prop-match-end loc)))
+          (remove-text-properties beg-match end-match '(lsp-semantic-token--comment-beg))
+          (cl-loop for i from beg-match below end-match do
```

Comment:
is it necessary to loop here (i.e., can't `beg-match` and `end-match` simply be passed to `{put,remove}-text-propert{y,ies}`?